toasty/samplers.py

```diff
@@ -431,6 +432,174 @@ def vec2pix(lon, lat):
     return vec2pix
 
 
+class WcsSampler(object):
+    """Create a sampler for data with a generic WCS specification.
+
+    The sampler returns values in float32, regardless of input data type.
+
+    Parameters
+    ----------
+    data : array
+        The image data.
+    wcs : :class:`astropy.wcs.WCS`
+        A WCS data structure describing the data.
+
+    Returns
+    -------
+    A function that samples the data; the call signature is
+    ``vec2pix(lon, lat)-> data``, where the inputs and output are 2D arrays
+    and *lon* and *lat* are in radians.
+    """
+
+    def __init__(self, data, wcs):
+        import numpy as np
+
+        self._image = np.float32(data)
+        self._wcs = wcs
+
+    def _image_bounds(self):
+        corners = self._wcs.calc_footprint()
+        lon_min = np.min(corners[:, 0]) * np.pi
+        lon_max = np.max(corners[:, 0]) * np.pi
+        lat_min = np.min(corners[:, 1]) * np.pi
+        lat_max = np.max(corners[:, 1]) * np.pi
+        return lon_min / 180, lon_max / 180, lat_min / 180, lat_max / 180
```

Comment:
This function fails badly with the DASCH sample file `rh04475_00_01ww_tnx.fit` because `calc_footprint()` only samples the corners of the image, while this image touches the north pole in its middle, which means that both the RA and Dec bounds are far far too small. I want to sit down and think about how to handle this.